### PR TITLE
Reviews and Pull Request comments

### PR DESCRIPTION
**Pull request by jonasfranz**
This should contribute to go-gitea/gitea#3748.

* Resolve conflicts
* Add review model
* Add comment types

This is WIP.

## Reviews
Reviews should bundle all code comments created for the initial review. All code comments created for this review are linked to the review model.

## Comment types
`CommentTypeCode`: A comment at a line of a code of a PR (UI not implemented)
`CommentTypeReview`: Notifies the participants about the review. (UI not implemented)

## TODO
- [x] Render code comments on the right place
- [x] Add support for comments on both sides
- [ ] Show code comments in the "conversation" after a review got submitted
- [ ] Show single code comment in the "conversation"
- [ ] Remove dummy data from "code comment form" (line, path, side)
- [x] Add support for "Add single comments"
- [ ] Order/Separate code comments by thread/review
- [x] ~~Blocked by go-gitea/git#116~~

### CODE REVIEW

File: models/review.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+const ReviewTypeUnknown ReviewType = -1
+
+const (
+	// ReviewTypeApprove approves changes
+	ReviewTypeApprove ReviewType = iota
+	// ReviewTypeComment gives general feedback
+	ReviewTypeComment
+	// ReviewTypeReject gives feedback blocking merge
+	ReviewTypeReject
+	// ReviewTypePending is a review which is not published yet
```
Comment (lafriks) [inline models/review.go]: It would be better that this would be moved as first

Comment (jonasfranz) [reply models/review.go]: @lafriks Done

**Comment by jonasfranz** [comment]
I've implemented the submit review form by implementing review comments:
review comments
I am not really contented with the icons. UI needs to be improved but won't in this PR.

Previously I've added a way of rendering comments in pull files code.
This is really buggy and I need some help with that @lafriks. 
At the moment it only works for the splitted diff view and takes up way to much space. In addition it is only shown on the right side every time.
comments on code

File: models/issue_comment.go [hunk 2]
```diff
@@ -788,3 +789,39 @@ func DeleteComment(comment *Comment) error {
...
+		IssueID: issue.ID,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if err = issue.loadRepo(e); err != nil {
+		return nil, err
+	}
+	for _, comment := range comments {
+		if err = comment.loadReview(e); err != nil && !IsErrReviewNotExist(err) {
```
Comment (lafriks) [inline models/issue_comment.go]: This should be remade to load reviews at once for all comments

Comment (jonasfranz) [reply models/issue_comment.go]: @lafriks Done

**Comment by jonasfranz** [comment]
@lafriks Now it works also for the unified view but adding comments will result in adding comments at the wrong place. This happens because the javascript part is not ready yet.

**Comment by jonasfranz** [comment]
Comments get invalidated now after the commented LOC get changed by a push to the head branch. That is implemented via git blame. ~~=> Blocked by go-gitea/git#116~~

**Comment by appleboy** [comment]
@JonasFranzDEV merged go-gitea/git#116

**Pull request merged**